frontend/src/lib/utils/sns-transactions.utils.ts

```diff
@@ -0,0 +1,204 @@
+import { toastsError } from "$lib/stores/toasts.store";
+import type { Account } from "$lib/types/account";
+import { TokenAmount } from "@dfinity/nns";
+import type { Principal } from "@dfinity/principal";
+import {
+  encodeSnsAccount,
+  type SnsTransaction,
+  type SnsTransactionWithId,
+} from "@dfinity/sns";
+import { fromNullable } from "@dfinity/utils";
+import type { SnsTransactionsStore } from "../stores/sns-transactions.store";
+import {
+  AccountTransactionType,
+  mapToSelfTransaction,
+  showTransactionFee,
+  type Transaction,
+} from "./transactions.utils";
+
+export interface SnsTransactionData {
+  toSelfTransaction: boolean;
+  transaction: SnsTransactionWithId;
+}
+
+/**
+ * Returns transactions of an SNS account sorted by date (newest first).
+ * Filters out duplicated transactions.
+ * A duplicated transaction is normally one made to itself.
+ * The data of the duplicated transaction is the same in both transactions. No need to show it twice.
+ *
+ * @param params
+ * @param {Account} params.account
+ * @param {Principal} params.rootCanisterId
+ * @param {SnsTransactionsStore} params.store
+ * @returns {SnsTransactionWithId[]}
+ */
+export const getSortedTransactionsFromStore = ({
+  store,
+  rootCanisterId,
+  account,
+}: {
+  store: SnsTransactionsStore;
+  rootCanisterId: Principal;
+  account: Account;
+}): SnsTransactionData[] =>
+  mapToSelfTransaction(
+    store[rootCanisterId.toText()]?.[account.identifier].transactions ?? []
+  ).sort(({ transaction: txA }, { transaction: txB }) =>
+    Number(txB.transaction.timestamp - txA.transaction.timestamp)
+  );
+
+/**
+ * Returns whether all transactions of an SNS account have been loaded.
+ *
+ * @param params
+ * @param {Account} params.account
+ * @param {Principal} params.rootCanisterId
+ * @param {SnsTransactionsStore} params.store
+ * @returns {boolean}
+ */
+export const isTransactionsCompleted = ({
+  store,
+  rootCanisterId,
+  account,
+}: {
+  store: SnsTransactionsStore;
+  rootCanisterId: Principal;
+  account: Account;
+}): boolean => store[rootCanisterId.toText()]?.[account.identifier].completed;
```

Comment:
Just because of `?.`
```suggestion
}): boolean => Boolean(store[rootCanisterId.toText()]?.[account.identifier].completed);
```

Reply:
Good point, TS should have complained...